recipes/scala/src/main/scala/com/adobe/platform/ml/helper/Helper.scala

```diff
@@ -54,18 +54,39 @@ class Helper {
     val apiKey: String = sparkSession.sparkContext.getConf.get("ML_FRAMEWORK_IMS_CLIENT_ID", "").toString
     val sandboxName: String = sparkSession.sparkContext.getConf.get("sandboxName", "").toString
 
+    println("sandboxName=" + sandboxName)
+
     val dataSetId: String = configProperties.get(taskId).getOrElse("")
 
-    // Load the dataset
-    var df = sparkSession.read.format(PLATFORM_SDK_PQS_PACKAGE)
-      .option(QSOption.userToken, userToken)
-      .option(QSOption.serviceToken, serviceToken)
-      .option(QSOption.imsOrg, orgId)
-      .option(QSOption.apiKey, apiKey)
-      .option(QSOption.mode, PLATFORM_SDK_PQS_INTERACTIVE)
-      .option(QSOption.datasetId, dataSetId)
-      .option(QSOption.sandboxName, sandboxName)
-      .load()
+    // Load the dataset with or without sandbox
+    var df = if(sandboxName.isEmpty()) {
+      println("Reading dataset using default sandbox")
+      sparkSession.read.format(PLATFORM_SDK_PQS_PACKAGE)
+        .option(QSOption.userToken, userToken)
+        .option(QSOption.serviceToken, serviceToken)
+        .option(QSOption.imsOrg, orgId)
+        .option(QSOption.apiKey, apiKey)
+        // Batch mode can handle much larger datasets but but has a few minute
+        // additional startup time required.  Interactive can be faster to read
+        // smaller amounts of data but can be subject to timeouts if the amount
+        // of data is too large.
+        // See https://docs.adobe.com/content/help/en/experience-platform/data-science-workspace/authoring/spark.html#reading-a-dataset
+        .option(QSOption.mode, PLATFORM_SDK_PQS_BATCH)
+        .option(QSOption.datasetId, dataSetId)
+        .load()
+    } else {
+      println("Reading dataset from sandbox")
+      sparkSession.read.format(PLATFORM_SDK_PQS_PACKAGE)
+        .option(QSOption.userToken, userToken)
+        .option(QSOption.serviceToken, serviceToken)
+        .option(QSOption.imsOrg, orgId)
+        .option(QSOption.apiKey, apiKey)
+        .option(QSOption.mode, PLATFORM_SDK_PQS_BATCH)
+        .option(QSOption.datasetId, dataSetId)
+        .option(QSOption.sandboxName, sandboxName)
+        .load()
+    }
```

Comment:
For DRY-ing this code- you can just use the `Reader` first and then `load()` later:
```
var dataFrameReader = sparkSession.read.format(PLATFORM_SDK_PQS_PACKAGE)
  .option(QSOption.userToken, userToken)
  .option(QSOption.serviceToken, serviceToken)
  .option(QSOption.imsOrg, orgId)
  .option(QSOption.apiKey, apiKey)
  .option(QSOption.mode, PLATFORM_SDK_PQS_BATCH)
  .option(QSOption.datasetId, dataSetId)

if(!sandboxName.isEmpty()) {
  dataFrameReader.option(QSOption.sandboxName, sandboxName)
}

var df = dataFrameReader.load()
  
```

Reply:
incorporated this for Scala.